backend/stakepoold/server.go

```diff
@@ -242,18 +324,119 @@ func runMain() int {
 	log.Infof("Connected to dcrd (JSON-RPC API v%s) on %v",
 		nodeVer.String(), curnet.String())
 
-	_, height, err := nodeConn.GetBestBlock()
+	curHash, curBlock, err := nodeConn.GetBestBlock()
 	if err != nil {
 		log.Errorf("unable to get bestblock from dcrd: %v", err)
 		return 8
```

Comment:
It really is time to make this function return and error instead of an int.  Then on the caller you do

if err := runMain(); err != nil {
  fmt.Fprintf(os.Stderr, "%v\n", err)
}

The numbers are duplicated and mean nothing.  This also guaranteed has some errors that aren't printed making debugging painful.